api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -479,6 +506,100 @@ public void getProgramByUuid_shouldFindObjectGivenValidUuid() {
+		assertEquals(programCount, programs.size());
+	}
+
+	/**
+	 * @see ProgramWorkflowService#unretireProgram(Program)
+	 */
+	@Test
+	public void unretireProgram_shouldUnRetireProgram() {
+		Program program = Context.getProgramWorkflowService().getProgram(1);
+		assertEquals(program.getRetired(), false);
+
+		Program retiredProgram = Context.getProgramWorkflowService().retireProgram(program,"for testing");
+		assertEquals(program.getRetired(), true);
+
+		Program unretireProgram = Context.getProgramWorkflowService().unretireProgram(retiredProgram);
+		assertEquals(unretireProgram.getRetired(), false);
+	}
+
+	/**
+	 * @see ProgramWorkflowService#saveConceptStateConversion(ConceptStateConversion)
+	 */
+	@Test(expected = APIException.class)
+	public void saveConceptStateConversion_shouldFaileIfRequiredParametersMissing() {
+		ConceptStateConversion conceptStateConversion = new ConceptStateConversion(1);
+		// all required parameters are missing
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
+
+		conceptStateConversion.setConcept(cs.getConcept(3));
+		// ProgramWorkflow, ProgramWorkflowState are missing
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
+
+		conceptStateConversion.setProgramWorkflow(pws.getWorkflow(1));
+		// ProgramWorkflowState is missing
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
+	}
+
+	/**
+	 * @see ProgramWorkflowService#saveConceptStateConversion(ConceptStateConversion)
+	 */
+	@Test
+	public void saveConceptStateConversion_shouldSaveIfAllRequiredParametersAreSatisfied() {
+		ConceptStateConversion conceptStateConversion = new ConceptStateConversion(1);
+		conceptStateConversion.setConcept(cs.getConcept(3));
+		conceptStateConversion.setProgramWorkflow(pws.getWorkflow(1));
+		conceptStateConversion.setProgramWorkflowState(pws.getWorkflow(1).getState(1));
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
+	}
+
+	/**
+	 * @see ProgramWorkflowService#getAllConceptStateConversions()
+	 */
+	@Test
+	public  void getAllConceptStateConversions_shouldReturnAll() {
+		List<ConceptStateConversion> conceptStateConversions=Context.getProgramWorkflowService().getAllConceptStateConversions();
+		assertEquals(1,conceptStateConversions.size());
+	}
+
+	/**
+	 * @see ProgramWorkflowService#getConceptStateConversion(Integer)
+	 */
+	@Test
+	public void getConceptStateConversion_shouldGetById(){
+		ConceptStateConversion conceptStateConversion = Context.getProgramWorkflowService().getConceptStateConversion(1);
+		assertNotNull(conceptStateConversion);
+	}
+
+	/**
+	 * @see ProgramWorkflowService#purgeConceptStateConversion(ConceptStateConversion)
+	 */
+	@Test
+	public void purgeConceptStateConversion_shouldPurge() {
+		ConceptStateConversion conceptStateConversion = Context.getProgramWorkflowService().getConceptStateConversion(1);
```

Comment:
use the test class field `pws` instead of taking the service again from the `Context`. this applies to all tests.